router/batchrouter/asyncdestinationmanager/sftp/sftp_test.go

```diff
@@ -178,20 +180,58 @@ var _ = Describe("SFTP test", func() {
 			ctrl := gomock.NewController(GinkgoT())
 			fileManager := mock_sftp.NewMockFileManager(ctrl)
 			defaultManager := newDefaultManager(fileManager)
-			fileManager.EXPECT().Upload(gomock.Any(), gomock.Any()).Return(nil)
+			now := time.Now()
+			filePath := fmt.Sprintf("/tmp/testDir1/destination_id_1_someJobRunId_1/file_%d_%02d_%02d_%d_1.csv", now.Year(), now.Month(), now.Day(), now.Unix())
+			fileManager.EXPECT().Upload(gomock.Any(), filePath).Return(nil)
 			manager := common.SimpleAsyncDestinationManager{UploaderAndTransformer: defaultManager}
+			payload := make(map[int64]stdjson.RawMessage)
+			id := int64(1014)
+			sampleData := map[string]interface{}{
+				"source_job_run_id": "someJobRunId_1",
+			}
+			rawMessage, _ := json.Marshal(sampleData)
+			payload[id] = rawMessage
+
+			id = int64(1016)
+			payload[id] = rawMessage
+
 			asyncDestination := common.AsyncDestinationStruct{
-				ImportingJobIDs: []int64{1014, 1015, 1016, 1017},
-				FileName:        filepath.Join(currentDir, "testdata/uploadDataRecord.txt"),
-				Destination:     &destinations[0],
-				Manager:         manager,
+				ImportingJobIDs:       []int64{1014, 1015, 1016, 1017},
+				FileName:              filePath,
+				Destination:           &destinations[0],
+				Manager:               manager,
+				OriginalJobParameters: payload,
+				CreatedAt:             now,
+				PartFileNumber:        1,
+				SourceJobRunID:        "someJobRunId_1",
 			}
 			expected := common.AsyncUploadOutput{
 				DestinationID:   "destination_id_1",
 				SucceededJobIDs: []int64{1014, 1015, 1016, 1017},
 				SuccessResponse: "File Upload Success",
 			}
+
+			if err := os.MkdirAll(filepath.Dir(filePath), os.ModePerm); err != nil {
+				fmt.Printf("Failed to create temporary directory: %v\n", err)
+				return
+			}
+
+			tempFile, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0o600)
+			if err != nil {
+				log.Fatalf("Failed to create temporary file: %v", err)
+			}
+			defer os.RemoveAll("/tmp/testDir1")
+
+			data := []byte(`{"message":{"action":"insert","fields":{"email":"john@email.com","name":"john"},"messageId":"b6543bc0-f280-4642-8176-4b8d5a1b8fb6","originalTimestamp":"2024-04-20T22:29:37.731+05:30","receivedAt":"2024-04-20T22:29:36.843+05:30","request_ip":"[::1]","rudderId":"853ae90f-0351-424b-973e-a615e6487517","sentAt":"2024-04-20T22:29:37.731+05:30","timestamp":"2024-04-20T22:29:36.842+05:30","type":"record"},"metadata":{"job_id":1}}`)
+			if _, err := tempFile.Write(data); err != nil {
+				log.Fatalf("Failed to write to temporary file: %v", err)
+			}
+
 			received := manager.Upload(&asyncDestination)
+			if err != nil {
```

Comment:
error handling should be next to the line where error is returned, is returning and error missed in the line above?

Reply:
this `err` check is redundant. manager.Upload is not returning the error.